File: 459. Repeated Substring Pattern/C++/Repeated Substring Pattern.cpp
class Solution {
public:
    bool repeatedSubstringPattern(string str) {
        for (int i = str.length() / 2; i > 0; i--) {
            if (str.length() % i == 0) {
                bool flag = false;
                for (int offset = i; offset < str.length(); offset += i) {
                    for (int index = 0; index < i; index++) {
                        if (str[index] != str[offset + index]) {
                            flag = true;
                            break;
                        }
                    }
                    if (flag) {
                        break;
                    }
                }
                if (!flag) {
                    return true;
                }
            }
        }
        return false;
    }
};

File: 459. Repeated Substring Pattern/Python/Repeated Substring Pattern.py
class Solution(object):
    def repeatedSubstringPattern(self, str):
        """
        :type str: str
        :rtype: bool
        """
        for i in range(len(str) // 2, 0, -1):
            if len(str) % i == 0:
                flag = False
                for offset in range(i, len(str), i):
                    for index in range(0, i):
                        if str[index] != str[offset + index]:
                            flag = True
                            break
                    if flag:
                        break
                if not flag:
                    return True
        return False
